### ISSUE TITLE
False postive UnneededRec with nested attribute sets

### ISSUE THREAD

**Issue by nothingelsematters** [Synthetica9/nix-linter#43]
Example:

```nix
rec {
  a = {
    a-b = "a - b";
  };
  c = {
    c-d = a.a-b;
  };
}
```

It is treated as unneeded `rec` now

I need this `rec` here, because otherwise it results in:  
`undefined variable 'a' at ...`

**Comment by avdv** [comment]
Here is another example which triggers this false positive:
```
rec {
  defaultApp = 1;
  apps.default = defaultApp;

  devShell =  2;
  devShells.default = devShell;
}
```
results in `Unneeded 'rec' on set at fp.nix:1:1-7:2`.

Interestingly, if one comments out either the `apps.default` or `devShells.default` line the error is gone.
